kolibri/plugins/learn/assets/src/state/actions/main.js

```diff
@@ -647,10 +646,9 @@ function setAndSaveCurrentExamAttemptLog(store, contentId, itemId, currentAttemp
       new Promise(resolve => {
         const log = Object.assign({}, newExamAttemptLog);
         store.dispatch('SET_EXAM_ATTEMPT_LOGS', {
-          // prettier-ignore
```

Comment:
Hrm, why is prettier not obeying the comment here?

These parentheses are necessary in order to allow this to be properly transpiled by buble.

Reply:
I removed the comment, since I didn't know that's why it was there.

In the buble demo page, it does seem to transpile this kind of object with lots of variable property names. Was this a limitation of an older version?

[example](https://buble.surge.sh/#const%20foo%20%3D%20'foo'%3B%0Aconst%20baz%20%3D%20'baz'%3B%0A%0Aconst%20bar%20%3D%20%7B%0A%20%5Bfoo%5D%3A%20%7B%0A%20%20%20%5Bbaz%5D%3A%201%2C%20%0A%20%7D%2C%0A%7D)

Reply:
This is currently breaking the build on this PR:
```
    ERROR in ./kolibri/plugins/learn/assets/src/state/actions/main.js
    Module parse failed: /home/travis/build/learningequality/kolibri/node_modules/buble-loader/index.js??ref--1!/home/travis/build/learningequality/kolibri/node_modules/eslint-loader/index.js??ref--8!/home/travis/build/learningequality/kolibri/kolibri/plugins/learn/assets/src/state/actions/main.js Unexpected token (643:53)
    You may need an appropriate loader to handle this file type.
    |   UserExamResource.clearCache();
    | 
    |   store.dispatch('SET_EXAM_ATTEMPT_LOGS', ( obj$1 = {, obj[itemId] = currentAttemptLog, obj )}, obj$1[contentId] = ( obj = {}, obj$1 ));
    |   var obj;
    |   var obj$1;
     @ ./kolibri/plugins/learn/assets/src/app.js 3:0-48
```
This is a direct result of the bad buble transpilation. Can you test your hypothesis by updating Buble to 0.18.0 and see if it fixes the build?

Reply:
I just reverted the change in main.js. A buble upgrade should probably be done separately.

I do notice in that log, that 
```javascript
store.dispatch('SET_EXAM_ATTEMPT_LOGS', 
( obj$1 = {, obj[itemId] = currentAttemptLog, obj )}, obj$1[contentId] = ( obj = {}, obj$1 ));

```

does look different than what the demo site gives:

```javascript
store.dispatch('SET_EXAM_ATTEMPT_LOGS',
( obj$1 = {}, obj$1[contentId] = (( obj = {}, obj[itemId] = log, obj )), obj$1 ));

```

Looks like it's just reordering the expressions. Must be pretty subtle..

Reply:
Yeah, it was a known issue in Buble (hence my knowing to do the extra parentheses here), so it's possible it's been fixed, which would be nice.